3p/recaptcha.js

```diff
@@ -89,9 +89,8 @@ export function initRecaptcha(recaptchaApiBaseUrl = RECAPTCHA_API_URL) {
   let dataObject;
   try {
     dataObject = parseJson(win.name);
-  } catch (e) {
-    throw new Error(TAG + ' Could not parse the window name.');
-  }
+  } catch (e) {}
+  devAssert(dataObject, '%s Could not parse the window name.', TAG);
```

Comment:
the fix should be in the test. try use `allowConsoleError`

Reply:
Please read error in PR message.

Reply:
Sorry I don't fully understand. Could you elaborate?

Reply:
The test expects an error message, i.e. it is already using `allowConsoleError`. Since `3p/recaptcha.js` uses `throw new Error` instead of `dev().assert` or similar, no error is generated in the console. This error is significant enough to report a real error, so `devAssert` seems reasonable to add here.

Reply:
I think we should remove the `allowConsoleError` from the test instead. 

`devAssert` is stripped in production builds and we probably want to keep this error message (vs. some JSON parsing error).

Reply:
Use `userAssert` maybe? We do want to have the error shown on prod build.

Reply:
Do callers of this function expect it to throw? `userAssert` would break that.

We can do `throw user().createEror(...)` if we want to classify this error differently.

Reply:
Thank you for the reviews! @choumx, when you say `devAssert` is stipped in production, do you just mean this `if` block?
```
export function devAssert(...) {
  if (getMode().minified) {
    return shouldBeTrueish;
  }
  ...
```
Or do you mean `devAssert()` and `dev().assert()` are really both ineffective in production? My hope was to just swap `devAssert()` for `dev().assert()` in my change. This would achieve the same `throw` from the original code (before my change) and a `console.error`+`reportError`.

Reply:
Yup, `dev().assert` has the same behavior actually. `devAssert` is the new hotness -- the only difference between it and `dev().assert` is better type inference, so `devAssert` should always be preferred.

`userAssert` is another option here since it's not stripped in minified/prod builds, so an error will still be thrown. Though the try/catch becomes a bit useless (we could remove it and use `tryParseJson` in that case).

Reply:
Thanks for the additional look. I'm impartial and happy to implement either change, just let me know which is preferred:
```
  dataObject = tryParseJson(win.name);
  userAssert(dataObject, '%s Could not parse the window name.', TAG);
```
and no change to associated test.

**or**

```
  try {
    dataObject = parseJson(win.name);
  } catch (e) {
    throw new Error(TAG + ' Could not parse the window name.');
  }
```
and remove `allowConsoleError` from associated test.

Reply:
I have a slight preference for the latter since `win.name` has some user-configurable data but is mostly an internal construction. I.e. a malformed name is probably our fault, not a user error.